Address PR71 refactor (small changes, additions) (#77)
zammad/zammad-admin-documentation: manage/roles/user-preferences-permissions.rst

>                                        .. note:: Agents only receive ticket notifications for
>                                           :doc:`groups they have “full” access to </manage/groups/access-levels>`.
> 
>                                        .. hint:: Customers don't need this permission.

Reviewer: Would prefer to combine these two notes, maybe like so?

```suggestion
                                       .. note:: Agents can only receive ticket notifications for
                                          :doc:`groups they have “full” access to </manage/groups/access-levels>`.
                                          Customers can’t receive ticket notifications at all.
```

Author: Thanks for your input, will be address with commit 229e5a8c55fcf709e0eadb232d165c1959a47a63

                                       .. warning:: 🔑 **Third-party authentication / LDAP users:**

                                          Be sure to **revoke this permission** for all your users.

>                                           Otherwise, users will have two passwords (local & LDAP).

Reviewer: Can you explain how this works? AFAIU Zammad only has one column in the users DB table for user passwords. How would users end up with two?

Author: Yes sure.
So basically Zammad doesn't require a password on your user account if you use a third party login source like LDAP.

When using LDAP, Zammad will contact your LDAP server during your authentication. It knows if the account has been synced from LDAP and thus will handle exactly that for your user. This is also the reason why the LDAP server has to be available during the moment you're trying to authenticate with a LDAP user.

Zammad does not (and never) synchronize the password of the LDAP user. 
So if I say you're having two passwords technically I do mean:

- you can have an independent, different password on the user account that lifes in Zammads database
- The LDAP source is used for your authentication try
- if you do have two passwords, one technically could fail but the other password may allow you to login
   - for instance: Your ldap server is not available due to network issues (this counts as authentication fail in terms of Zammad) - BUT if you have a local password on that user within the Zammad databae and it's correct, you're still able to login

I hope this helps to understand the scope better.

Reviewer: Ack! Totally didn't realize this was sitting here waiting for a response from me. 

Great explanation. I'd like to finesse this explanation a little more, because it's technically correct, but creates more questions than it answers:

* "will users always have two passwords, or is it just that they _can_ have two passwords?"
* "are the two passwords necessarily different? what happens when they are the same?"
* "will both passwords work?"

(Of course, your explanation answered these questions for me, but we want to keep it as short as possible.)

I think a higher-level explanation of the **intent** / security model would work better:

```suggestion
                                          When using a third-party identity server (like LDAP),
                                          the whole point is to let *it* take care of authentication
                                          so that passwords never have to live in Zammad’s database.
```

Author: Ah good point. This part gently gives nose bleeding. :D

Thanks!
